### PR TITLE
HEEDLS-841 Deactivate admin when deletion fails

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-841

### Description
Fall back to deactivating the admin account when an exception occurs when attempting to delete the account.

### Screenshots
N/A

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

**Review by livzorn — APPROVED** [review]
Looks good to me!

File: DigitalLearningSolutions.Data/Services/UserService.cs [hunk 1]
```diff
@@ -437,7 +441,15 @@ public void DeactivateOrDeleteAdmin(int adminId)
...
             else
             {
-                userDataService.DeleteAdminUser(adminId);
+                try
+                {
+                    userDataService.DeleteAdminUser(adminId);
+                }
+                catch (Exception ex)
+                {
+                    logger.LogWarning(ex, $"Error attempting to delete admin {adminId} with no sessions, deactivating them instead.");
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Data/Services/UserService.cs:450]: Do we want to include the exception message in log message on the off chance something else has occurred?

**Review by DanBloxham-sw — COMMENTED** [review]
Looks good, minor question about the log message.

File: DigitalLearningSolutions.Data/Services/UserService.cs (hunk 1 above)
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data/Services/UserService.cs:450]: With this method, the message will be logged to the Message and MessageTempate columns, and the exception to the Exception column

**Review by DanBloxham-sw — APPROVED** [review]

**Pull request merged**